gen-compdocs/generators/doc.go

```diff
@@ -54,21 +54,41 @@ guide. You can file document formatting bugs against the
 func GenMarkdownTree(cmd *cobra.Command, dir string, withTitle bool) error {
 	identity := func(s string) string { return s }
 	emptyStr := func(s string) string { return "" }
-	return GenMarkdownTreeCustom(cmd, dir, emptyStr, identity, withTitle)
+	return GenMarkdownTreeCustom(cmd, dir, "", emptyStr, identity, withTitle)
 }
 
-func GenMarkdownTreeCustom(cmd *cobra.Command, dir string, filePrepender, linkHandler func(string) string, withTitle bool) error {
+func GenMarkdownTreeCustom(cmd *cobra.Command, dir string, subdir string, filePrepender, linkHandler func(string) string, withTitle bool) error {
 	for _, c := range cmd.Commands() {
 		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
 			continue
 		}
-		if err := GenMarkdownTreeCustom(c, dir, filePrepender, linkHandler, withTitle); err != nil {
+		// CommandPath example: 'kubectl top pod'
+		parts := strings.Split(c.CommandPath(), " ")
+		subdir := ""
+		if len(parts) > 1 {
+			subdir = parts[0] + "_" + parts[1]
```

Comment:
In case there are more parts, would it be better to do:

```go
subdir = strings.Join(parts, "_")
```

I can't think of a specific case right now, but I believe it's possible some commands will have more parts.

Reply:
Maybe we don't want the tree structure to be too deep?
(a comment explaining intent would help a lot here)